src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,206 @@
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isFormOpen = useLocation().pathname === '/people/new';
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const appliedQuery = searchParams.get('query') || '';
+  const sortBy = searchParams.get('sortBy') || '';
+  const sortOrder = searchParams.get('sortOrder') || '';
+
+  const [query, setQuery] = useState(appliedQuery);
+
+  useEffect(() => {
+    loadData()
+      .then(loadedPeople => setPeople(loadedPeople))
+      .catch(() => setIsError(true));
+  }, []);
+
+  const applyQuery = useCallback(
+    debounce((newQuery: string) => {
+      setSearchParams(getSearchWith({ query: newQuery }, searchParams));
+    }, 500),
+    [],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleNewPersonSubmit = (newPerson: Person) => {
+    setPeople(prevPeople => {
+      return ([
+        ...prevPeople,
+        linkPersonWithParents(newPerson, prevPeople),
+      ]);
+    });
+  };
+
+  const preparePeople = () => {
+    const preparedPeople = people.filter(person => (
+      person.name.toLowerCase()
+        .includes(appliedQuery.toLowerCase())
+      || person.motherName?.toLowerCase()
+        .includes(appliedQuery.toLowerCase())
+      || person.fatherName?.toLowerCase()
+        .includes(appliedQuery.toLowerCase())
+    ));
+
+    const numOrder = sortOrder === 'asc'
+      ? 1
+      : -1;
+
+    switch (sortBy) {
+      case PersonEnum.Name:
+      case PersonEnum.Sex:
+        preparedPeople.sort((firstPerson, secondPerson) => (
+          firstPerson[sortBy].localeCompare(secondPerson[sortBy]) * numOrder
+        ));
+        break;
+
+      case PersonEnum.Born:
+      case PersonEnum.Died:
+        preparedPeople.sort((firstPerson, secondPerson) => (
+          (firstPerson[sortBy] - secondPerson[sortBy]) * numOrder
+        ));
+        break;
+
+      default:
+        break;
+    }
+
+    return preparedPeople;
+  };
+
+  const preparedPeople = preparePeople();
+
+  return (
+    <div className="PeoplePage">
+      <h1 className="text-center mb-3">People Page</h1>
+
+      <div className="row mb-3">
+        <div className="col">
+          <div className="form-floating">
+            <input
+              type="text"
+              id="nameInput"
+              className="form-control"
+              placeholder="Name"
+              value={query}
+              onChange={handleInputChange}
+              data-cy="filterInput"
+            />
+
+            <label htmlFor="nameInput">Name</label>
+          </div>
+        </div>
+
+        <div className="col">
+          {isFormOpen
+            ? (
+              <NewPerson
+                people={people}
+                onNewPersonSubmit={handleNewPersonSubmit}
+              />
```

Comment:
Let's render this Form on a separate route:
<img width="656" alt="image" src="https://user-images.githubusercontent.com/52370890/183680093-b51b8fa6-b359-4aad-8b71-14a6e5b0ffd9.png">